### PR TITLE
Line wrapping

### PR DESCRIPTION
**Pull request by nilnor**
This adds line wrapping support for aullar.  It supports the previously existing `line_wrapping` configuration variable, which supports wrapping on either character or word (or none). It also introduces two new configuration variables related to wrapping:

- `line_wrapping_navigation`

This provides control over how wrapped lines are navigated. "real" means cursor movement will step over the line in an ordinary fashion even if it's wrapped, and "visual" means that cursor movement will walk into a wrapped line. The mode is also reflected in the current line marker which will highlight the entire line for the former, and the visual line for the latter.

- `line_wrapping_symbol`

This is the symbol used for indicating wrapping at the right side of the wrapped visual line. The appearance of the symbol itself is controlled using ordinary styles.

The one (big) thing that is not present is "proper" scrolling of lines when the navigation mode is set to visual. We currently only support scrolling by real lines, meaning scrolling can now appear a bit jagged since scrolling one line might mean actually shifting the screen by two or more visual lines. The plan is not to include that in this PR however, but instead leave that for the  future.

### CODE REVIEW

File: lib/howl/ui/editor.moon [hunk 1]
```diff
@@ -169,21 +169,20 @@ class Editor extends PropertyObject
...
-    get: =>
-      -- sci_val = @sci\get_wrap_mode!
-      -- switch sci_val
-      --   when Scintilla.SC_WRAP_NONE then 'none'
-      --   when Scintilla.SC_WRAP_WORD then 'word'
-      --   when Scintilla.SC_WRAP_CHAR then 'character'
-      --   else '(unknown)'
+    get: => @view.config.view_line_wrap
+    set: (value) =>
+      unless value\umatch(r'^(?:none|word|character)$')
```
Comment (kirbyfan64) [inline lib/howl/ui/editor.moon]: Particular reason for the parens here? Seems slightly inconsistent with the rest of the file.

**Comment by kirbyfan64** [comment]
So, while toying with this, I noticed that it's SUPER hard to select the very last character before the line wraps.

Still great though! :)

File: lib/howl/ui/editor.moon (hunk 1 above)
Comment (nilnor) [inline lib/howl/ui/editor.moon]: No reason, those just slunk in there (removed now).

**Comment by nilnor** [comment]
@kirbyfan64 using the mouse presumably? I just tried that (for the first time apparently) and that is indeed not working as it should.

**Pull request merged**